### PR TITLE
Improve SCM info in build scans

### PR DESCRIPTION
**Pull request by mark-vieira**
This PR removes usage of the Nebula SCM plugin info for decorating build scans. This information tended to be quite inaccurate in many scenarios so instead we just use known Jenkins environment variables for these things. We also include some additional information exposed by Jenkins for pull requests such as the PR number, target branch and link back to the original PR.

### CODE REVIEW

File: gradle/build-scan.gradle [hunk 1]
```diff
@@ -18,26 +18,32 @@ buildScan {
             value 'Jenkins Worker Label', it
         }
 
+        def isPrBuild = System.getenv('ROOT_BUILD_CAUSE_GHPRBCAUSE') != null
```
Comment (rjernst) [inline gradle/build-scan.gradle]: This can be boolean, no need for def?

File: gradle/build-scan.gradle [hunk 2]
```diff
@@ -18,26 +18,32 @@ buildScan {
...
+        // Add SCM information
+        if (isPrBuild) {
+            value 'Git Commit ID', System.getenv('ghprbActualCommit')
+            value 'Git Branch', System.getenv('ghprbTargetBranch')
+            tag System.getenv('ghprbTargetBranch')
+            tag "pr/${System.getenv('ghprbPullId')}"
+            link 'Source', "https://github.com/elastic/elasticsearch/tree/${System.getenv('ghprbActualCommit')}"
+            link 'Pull Request', System.getenv('ghprbPullLink')
+        } else {
+            def branch = System.getenv('GIT_BRANCH').split('/').last()
```
Comment (rjernst) [inline gradle/build-scan.gradle]: This can be String, no need for def?

**Review by rjernst — APPROVED** [review]
Lgtm

**Review by alpar-t — APPROVED** [review]
LGTM!

**Pull request merged**